llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/VectorExt/IR/VectorExtBase.td

```diff
@@ -47,11 +83,15 @@ def PerDimLayoutAttr : IREEVectorExt_Attr<"PerDimLayout"> {
     we can reason about layouts, propagating layouts and layout conflicts.
    }];
    let parameters = (ins
-     ArrayRefParameter<"std::string", "labels for the high dimensional layout dims">:$labels,
+     ArrayRefParameter<"LayoutDimensionAttr", "labels for the high dimensional layout dims">:$labels,
      ArrayRefParameter<"int64_t", "shapes for the high dimensional layout dims">:$shapes
    );
-   let hasCustomAssemblyFormat = 1;
+   let assemblyFormat = "`<` `[` $labels `]` `,` `[` $shapes `]` `>`";
```

Comment:
I noticed that while printing this, the printer inserts an extra space before the first label. Can we fix that?